File: 1-two-sum/1-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    
    let hm = new Map();
    let result = [];
    
    for(let i=0; i <= nums.length; i++){
        if(hm.has(target - nums[i])){
            return [hm.get(target - nums[i]), i]
        }
        
        else{
            hm.set(nums[i], i)
        }
            
    };

};

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        hm = {}
        
        for i in range(len(nums)):
            j = target - nums[i]
            if j in hm:
                return [i, nums.index(j)]
            else:
                hm[nums[i]] = True